File: Design Patterns/Strategy Pattern/Java/com.faculdadevincit.designpatterns.strategypattern/src/PagamentoCredito.java
public class PagamentoCredito implements IPagamentoStrategy{

    @Override
    public void pagar(double valor) {
        double acrescimo = valor *0.10;
        double valorComAcrescimo = valor + acrescimo;
        System.out.println("Pagamento de R$" + valorComAcrescimo + " em Cartão de Crédito realizado com sucesso!");
        System.out.println("Acréscimo no valor de R$" + acrescimo);
    }

}

File: Design Patterns/Strategy Pattern/Python/pagamento_credito.py
from pagamento_strategy import PagamentoStrategy


class PagamentoCredito(PagamentoStrategy):
    def pagar(self, valor):
        acrescimo = valor*0.10
        valorComAcrescimo = valor + acrescimo
        print(f"Pagamento de R${valorComAcrescimo} em cartão de crédito realizado com sucesso!")
        print(f"Valor do acréscimo do crédito R${acrescimo}")